src/iceberg/catalog/memory/in_memory_catalog.cc

```diff
@@ -314,6 +324,13 @@ Result<std::string> InMemoryNamespace::GetTableMetadataLocation(
   return it->second;
 }
 
+Status InMemoryNamespace::UpdateTableMetadataLocation(
+    TableIdentifier const& table_ident, std::string const& metadata_location) {
```

Comment:
```suggestion
    const TableIdentifier& table_ident, const std::string& metadata_location) {
```
I think it's better to keep the same style `const T&` as function declaration.